client/modules/IDE/components/show-hint.js

```diff
@@ -0,0 +1,718 @@
+    },
+
+    pick: function (data, i) {
+      var completion = data.list[i],
+        self = this;
+      this.cm.operation(function () {
+        if (completion.hint) completion.hint(self.cm, data, completion);
+        else
+          self.cm.replaceRange(
+            getText(completion),
+            completion.from || data.from,
+            completion.to || data.to,
+            'complete'
+          );
+        CodeMirror.signal(data, 'pick', completion);
+        self.cm.scrollIntoView();
+      });
+      if (this.options.closeOnPick) {
+        this.close();
+      }
+    },
+
+    cursorActivity: function () {
+      if (this.debounce) {
+        cancelAnimationFrame(this.debounce);
+        this.debounce = 0;
+      }
+
+      var identStart = this.startPos;
+      if (this.data) {
+        identStart = this.data.from;
+      }
+
+      var pos = this.cm.getCursor(),
+        line = this.cm.getLine(pos.line);
+      if (
+        pos.line != this.startPos.line ||
+        line.length - pos.ch != this.startLen - this.startPos.ch ||
+        pos.ch < identStart.ch ||
+        this.cm.somethingSelected() ||
+        !pos.ch ||
+        this.options.closeCharacters.test(line.charAt(pos.ch - 1))
+      ) {
+        this.close();
+      } else {
+        var self = this;
+        this.debounce = requestAnimationFrame(function () {
+          self.update();
+        });
+        if (this.widget) this.widget.disable();
+      }
+    },
+
+    update: function (first) {
+      if (this.tick == null) return;
+      var self = this,
+        myTick = ++this.tick;
+      fetchHints(this.options.hint, this.cm, this.options, function (data) {
+        if (self.tick == myTick) self.finishUpdate(data, first);
+      });
+    },
+
+    finishUpdate: function (data, first) {
+      if (this.data) CodeMirror.signal(this.data, 'update');
+
+      var picked =
+        (this.widget && this.widget.picked) ||
+        (first && this.options.completeSingle);
+      if (this.widget) this.widget.close();
+
+      this.data = data;
+
+      if (data && data.list.length) {
+        if (picked && data.list.length == 1) {
+          this.pick(data, 0);
+        } else {
+          this.widget = new Widget(this, data);
+          CodeMirror.signal(data, 'shown');
+        }
+      }
+    }
+  };
+
+  function parseOptions(cm, pos, options) {
+    var editor = cm.options.hintOptions;
+    var out = {};
+    for (var prop in defaultOptions) out[prop] = defaultOptions[prop];
+    if (editor)
+      for (var prop in editor)
+        if (editor[prop] !== undefined) out[prop] = editor[prop];
+    if (options)
+      for (var prop in options)
+        if (options[prop] !== undefined) out[prop] = options[prop];
+    if (out.hint.resolve) out.hint = out.hint.resolve(cm, pos);
+    return out;
+  }
+
+  function getText(completion) {
+    if (typeof completion == 'string') return completion;
+    else return completion.item.text;
```

Comment:
Let me know if this might've been an issue on my end, but this feature wouldn't work for me until `completion.item.text` was updated to `completion.text`!

Reply:
Hi @raclim thanks for trying it out. I couldn't replicate or locate the issue. The only difference that I can think of between our environments is the `node_modules` folder, but you can also see that the only changes in the `package-lock.json` are all about the `fuse.js` package. Nevertheless, if you could try a clean reinstall of the modules, it might help.

The interface looks like the version for CSS (activated for `.css` files) - no links to the references, just completions. But it shouldn't be the case when a JS file is activated. I tried to change some code to ensure the correct mode is used, but don't know if it works as I cannot replicate the problem locally.

Would it be possible for you to create a release environment like [this](https://github.com/processing/p5.js-web-editor/pull/1873#issuecomment-881003367)? That would also be very helpful!

Reply:
Thanks for those suggestions, you were right I was behind a few packages 😂 Works great for me now!